flake.nix

```
@@ -29,21 +29,25 @@
         ;
 
       importPackages = pkgs: let
-        # nixosTests is overriden with tests defined in this
-        # flake.
+        # nixosTests is overriden with tests defined in this flake.
         nixosTests =
           pkgs.nixosTests
           // (
             let
               dir = ./tests;
-            in
-              mapAttrs (name: _:
-                pkgs.nixosTest (import (dir + "/${name}") {
+              testDirs = readDir dir;
+              dirToTest = name: _: let
+                mkNetwork = import "${dir}/${name}";
+                network = mkNetwork {
                   inherit pkgs;
                   inherit (pkgs) lib;
                   modules = extendedModules;
                   configurations = importNixosConfigurations;
-                })) (readDir dir)
+                };
+              in
+                pkgs.nixosTest network;
```

I am not sure that the argument passed to `nixosTest` should be called a "network". It does contain `nodes` that will end up networked, but it also contains the test script itself, and some other metadata (test name) and configuration (for interactive testing). The [NixOS manual says]: "We refer to the whole test above as a test module, whereas the values in [`nodes.<name>`](https://nixos.org/manual/nixos/stable/#test-opt-nodes) are NixOS modules themselves."

So, I'd go for `testModule` instead.
```suggestion
                pkgs.nixosTest testModule;
```

---

I agree.